src/Sylius/Bundle/ShopBundle/Controller/ContactController.php

```diff
@@ -0,0 +1,102 @@
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ShopBundle\Controller;
+
+use Sylius\Bundle\CoreBundle\EmailManager\ContactEmailManager;
+use Sylius\Bundle\CoreBundle\Form\Type\ContactType;
+use Sylius\Component\Channel\Context\ChannelContextInterface;
+use Symfony\Bundle\FrameworkBundle\Controller\Controller;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+
+/**
+ * @author Grzegorz Sadowski <grzegorz.sadowski@lakion.com>
+ */
+class ContactController extends Controller
+{
+    /**
+     * @var ChannelContextInterface
+     */
+    private $channelContext;
+
+    /**
+     * @var ContactEmailManager
+     */
+    private $contactEmailManager;
+
+    /**
+     * @param ChannelContextInterface $channelContext
+     * @param ContactEmailManager $contactEmailManager
+     */
+    public function __construct(
+        ChannelContextInterface $channelContext,
+        ContactEmailManager $contactEmailManager
+    ) {
+        $this->channelContext = $channelContext;
+        $this->contactEmailManager = $contactEmailManager;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function requestAction(Request $request)
+    {
+        $formType = $this->getSyliusAttribute($request, 'form', ContactType::class);
+        $form = $this->createForm($formType);
+
+        if ($request->isMethod('POST') && $form->handleRequest($request)->isValid()) {
+            $data = $form->getData();
+            $channel = $this->channelContext->getChannel();
+
+            try {
+                $this->contactEmailManager->sendContactRequest($data, $channel);
+            } catch (\InvalidArgumentException $exception) {
+                $errorMessage = $this->getSyliusAttribute($request, 'error_flash', 'sylius.contact.request_error');
+                $this->addFlash('error', $errorMessage);
+
+                return $this->redirect($request->headers->get('referer'));
+            }
+
+            $successMessage = $this->getSyliusAttribute($request, 'success_flash', 'sylius.contact.request_success');
+            $this->addFlash('success', $successMessage);
+
+            $redirectRoute = $this->getSyliusAttribute($request, 'redirect', 'referer');
+
+            return $this->redirectToRoute($redirectRoute);
+        }
+
+        $template = $this->getSyliusAttribute($request, 'template', "@SyliusShop/Contact/request.html.twig");
+
+        return $this->render(
+            $template,
+            [
+                'form' => $form->createView(),
+            ]
+        );
+    }
+
+    /**
+     * @param Request $request
+     * @param string $attribute
+     * @param mixed $default
+     *
+     * @return mixed
+     */
+    private function getSyliusAttribute(Request $request, $attribute, $default = null)
```

Comment:
Maybe something like `getFormTypeName`, and you should probably throw exception if you depend on this parameter.